contracts/SpokePoolV3Periphery.sol

```diff
@@ -311,18 +191,35 @@ contract SpokePoolV3Periphery is Lockable, MultiCaller {
             r,
             s
         );
-        IERC20 _swapToken = IERC20(address(swapToken)); // Cast IERC20Auth to IERC20.
-
-        _swapAndBridge(
+        _callSwapAndBridge(
+            IERC20(address(swapToken)),
+            acrossInputToken,
+            exchange,
             routerCalldata,
             swapTokenAmount,
             minExpectedInputTokenAmount,
-            depositData,
-            _swapToken,
-            acrossInputToken
+            depositData
         );
     }
 
+    /**
+     * @notice Deposits an ERC20 token into the Spoke Pool contract.
+     * @dev User should probably just call SpokePool.depositV3 directly to save marginal gas costs,
+     * but this is added here for convenience in case caller only wants to interface Across through a single
+     * address.
+     * @param acrossInputToken ERC20 token to deposit.
+     * @param acrossInputAmount Amount of the input token to deposit.
+     * @param depositData Specifies the Across deposit params to send.
+     */
+    function depositERC20(
```

Comment:
Do we need this? Can we just call the spoke pool directly and save the extra gas?

Reply:
removed